components/MainTiles/MainTiles.tsx

```diff
@@ -29,22 +29,26 @@ export const MainTiles = memo<MainTilesProps>((props) => {
   });
 
   return (
-    <section className={styles.section}>
-      <h2 className={styles.heading}>Wszystkie artykuły</h2>
-      <div className={styles.container}>
-        <Link href="/zglos-serwis" passHref>
-          <Button as="a" icon="icon-plus">
-            Zgłoś serwis
-          </Button>
-        </Link>
-        <DisplayStyleSwitch value={props.displayStyle} onChange={changeDisplayStyle} />
-      </div>
+    <Section
+      title="Wszystkie artykuły"
+      buttons={
+        <>
+          <Link href="/zglos-serwis" passHref>
+            <Button as="a" icon="icon-plus">
+              Zgłoś serwis
+            </Button>
+          </Link>
+          <DisplayStyleSwitch value={props.displayStyle} onChange={changeDisplayStyle} />
+        </>
+      }
+    >
```

Comment:
Czy ta abstrakcja jest potrzebna? Moim zdaniem w tym miejscu preferujemy powtórzenie zamiast abstrakcji, imo będzie to bardziej czytelne. Dochodzi też kwestia przyszłych zmian. Wywaliłbym atrybut buttons i przekazywał Buttony w children.

Reply:
Popieram. Zresztą tutaj nadal powtarzamy większość kodu, a do tego jest on dłuższy.